### PR TITLE
Added some features for v3.0.0

### PR DESCRIPTION
**Pull request by indieSoftware**
- Added `Optional+Stringified`.
- Removed `UIDevice` properties for `isIPhoen`, `isIPad` and `isMac`.
- Added `UIDeviceProvider` to solve `UIDevice` dependencies.
- Improved the example app.
- Improved comments and documentation.

### CODE REVIEW

File: Sources/INCommonsExample/Screens/ViewConditions.swift [hunk 1]
```diff
@@ -2,27 +2,46 @@ import INCommons
...
+			Text("if modifier")
 			Button(action: {
-				self.isChanged.toggle()
+				toggleButton()
 			}, label: {
-				Text(strTitle)
+				Text(buttonTitle)
 					.padding(.all)
-					.if(isChanged == true) { $0.background(Color.green) }
+					.if(isToggled == true) { $0.background(Color.green) }
```
Comment (huseyinbagana) [inline Sources/INCommonsExample/Screens/ViewConditions.swift:22]:
```suggestion
					.if(isToggled) { $0.background(Color.green) }
```

File: Sources/INCommonsExample/Screens/ViewConditions.swift [hunk 2]
```diff
@@ -2,27 +2,46 @@ import INCommons
...
-				Text(strTitle)
+			Spacer().frame(height: 20)
+
+			Text("if-else modifier")
+			Button(action: {
+				toggleButton()
+			}, label: {
+				Text(buttonTitle)
+					.padding(.all)
+					.if(isToggled == true) { $0.background(Color.green) } else: { $0.background(Color.yellow) }
```
Comment (huseyinbagana) [inline Sources/INCommonsExample/Screens/ViewConditions.swift:33]:
```suggestion
					.if(isToggled) { $0.background(Color.green) } else: { $0.background(Color.yellow) }
```

**Review by huseyinbagana — CHANGES_REQUESTED** [review]

File: Tests/INCommonsTests/Swift/Optional/OptionStringifiedTests.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		let optionalString: String? = nil
+		let result = optionalString.stringified()
+		XCTAssertEqual("", result)
+	}
+
+	func testCustomFallbackStringCanBeProvided() throws {
+		let optionalString: String? = nil
+		let result = optionalString.stringified("Fallback")
+		XCTAssertEqual("Fallback", result)
+	}
```
[the remark this answers is not in the document]
Comment (indieSoftware) [reply Tests/INCommonsTests/Swift/Optional/OptionStringifiedTests.swift:26]: very good point! 👍 done

File: ReleaseSteps.md [hunk 4]
```diff
@@ -4,7 +4,10 @@
...
 	- Verify that all changes have been logged to [Changelog.md](Changelog.md).
 	- Verify that all tests passes.
-3. Run the `Scripts/updateVersion.sh` script and pass the new marketing version (e.g. `1.2.3`) as argument to prepare a new release version.
+3. Update build and marketing version.
+	- Because of a bug in agvtool and projects created with Xcode 13 we can't use the following script. We have to update the marketing version of the library manually. 
+	 - (Run the `Scripts/updateVersion.sh` script and pass the new marketing version (e.g. `1.2.3`) as argument to prepare a new release version.)
+  - However, we can call `Scripts/incrementBuildVersion.sh` to increment the lib's build version.
 4. Run the `Scripts/generateDocs.sh` script to update the documentation.
 5. Merge branch with master and push to origin.
 6. Run the `Scripts/releaseVersion.sh` script to create and push a new git tag for the lib's new version and to update cocoapods.
```
[the remark this answers is not in the document]
Comment (indieSoftware) [reply ReleaseSteps.md]: I have updated the description as well as the release script. Please review them again.

**Review by huseyinbagana — APPROVED** [review]

**Review by huseyinbagana — APPROVED** [review]

**Review by huseyinbagana — APPROVED** [review]

**Pull request merged**